src/main/java/org/terasology/module/health/components/RegenComponent.java

```diff
@@ -57,4 +57,10 @@
      */
     @Replicate
     public Map<Name, Instant> actions = new HashMap<>();
+
+    @Override
+    public void copy(RegenComponent other) {
+        this.remainder = other.remainder;
+        this.actions = other.actions;
```

Comment:
Both `Name` and `Instant` are immutable, so its probably sufficient to do a shallow clone of the map only?